.travis.yml

```diff
@@ -15,11 +15,24 @@ before_install:
 
 script:
     - find \( -name \*.c -o -name \*.h \) -exec sh -c '! grep --color "^[[:space:]]*//" $@' find {} \+ && echo No line comments found...
-    - docker run -v $TRAVIS_BUILD_DIR:/source ${OS_TYPE} bash -c "set -e;
-        ${INSTALL_REQUIREMENTS};
+    - echo > build.sh "set -e;
         cd /source;
         export CCFLAGS='-fno-strict-aliasing -Wno-unknown-pragmas -Wno-missing-braces -Wno-unused-result -Wno-return-type -Wno-int-to-pointer-cast -Wno-parentheses -Wno-unused -Wno-unused-but-set-variable -Wno-cpp -P';
         export CC=gcc;
         ./bin/package;
-        ./bin/package make -S 2>&1 | tee out.log ;
-        grep -q 'error:' out.log && exit 1 || exit 0;"
+        ./bin/package make ast-ksh 2>&1 | tee out.log;
+        grep -q 'error:' out.log && exit 1;
+        ln -s /usr/bin/cat arch/$(sh bin/package)/bin/;
+        ln -s /usr/bin/tee arch/$(sh bin/package)/bin/;
+        ln -s /usr/bin/chmod arch/$(sh bin/package)/bin/;
+        ulimit -n 1024;
+        ./bin/package test ksh93 2>&1 | tee test.log;
+        grep -q 'failed' test.log && exit 1 || exit 0;"
```

Comment:
This seems fragile. I wouldn't expect the word `failed` to appear in another context in the test output but you never know. Perhaps something like `grep -q '^test [^ ]* failed'` to help ensure it only matches entries like this one (from running the tests on my Ubuntu server):

```
test pty failed with exit code 1 [ 20 tests 1 error ]
```